File: backend/src/main/java/com/example/starter_project_2025/system/assessment/entity/AssessmentQuestion.java
package com.example.starter_project_2025.system.assessment.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.experimental.FieldDefaults;

import java.util.UUID;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@Table(name = "assessment_questions")
public class AssessmentQuestion {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    UUID id;

    @ManyToOne
    @JoinColumn(name = "assessment_id", nullable = false)
    @JsonIgnoreProperties({ "assessmentQuestions", "handler", "hibernateLazyInitializer" })
    Assessment assessment;

    @ManyToOne
    @JoinColumn(name = "question_id", nullable = false)
    @JsonIgnoreProperties({ "handler", "hibernateLazyInitializer","assessmentQuestions" })
    Question question;

    @Column(name = "score")
    Double score;

    @Column(name = "order_index")
    Integer orderIndex;
}

File: frontend/src/types/features/assessment/assessment-question.ts
export interface AssessmentQuestion {
    id: string;
    assessmentId: string;
    questionId: string;
    score: number;
    orderIndex: number;
}

export interface AssessmentQuestionCreateRequest {
    assessmentId: string;
    questionId: string;
    score: number;
    orderIndex: number;
}

export interface AssessmentQuestionUpdateRequest {
    score: number;
    orderIndex: number;
}
